### ISSUE TITLE
Unable to reproduce ACDC/Synapse results - seeking guidance

### ISSUE THREAD

**Issue by KyriakiKolpetinou** [OSUPCVLab/SegFormer3D#29]
Hi, 

I've been trying to reproduce the results reported in your paper for the ACDC and Synapse datasets using the nnFormer training pipeline as described, but I'm getting significantly lower Dice scores. My setup:

- nnFormer repo 
- Your SegFormer3D baseline integrated into nnFormer
- GPU: nvidia A100
- Patch size used: [14, 160, 160] for ACDC (from nnFormer plans) ((64,128,128) for Synapse)
- I have altered the Segformer network to accept non cubic inputs.
- I've also tried adamW, warmp and scheduling in place of SGD nnFormer uses but still cant get the correct results.

for example for ACDC I'm getting:
Dice_rv 0.8156461043024315
Dice_myo 0.8140759288442456
Dice_lv 0.9206212720308106

which differ a lot from those on the paper. Maybe I'm missing something. 

So my question is,
Are there any specific hyperparameters or training configurations beyond the default nnFormer settings that you used? Are the patch sizes im using wrong maybe?

Thank you in advance!

**Comment by KyriakiKolpetinou** [comment]
Hello,

Thank you for your answer. I, too, used the nnFormer repo unchanged, same pre processing, same lr, batch size, patch size, etc. The splits for train-val I took from the nnFromer json files. This is the model I used, the official Segformer if I'm not mistaken, only change that it accepts non cubic inputs. 
[segformer3d_baseline.py](https://github.com/user-attachments/files/24457542/segformer3d_baseline.py)

**Comment by duydngn89** [comment]
Hello,

Since you used the same nnFormer preprocessing, learning rate, batch size, patch size, and train/val splits, the difference is probably not from the training pipeline.

I have checked the segformer3d_baseline.py model. The main issue seems to be architectural. From the SegFormer3D file, your encoder uses isotropic downsampling, with the first stride effectively reducing all three dimensions. For ACDC, the input is highly anisotropic/non-cubic, with a small depth dimension.

I have already fixed that. You can email me at duy.nguyend8998@hcmut.edu.vn for a quick discussion, and I can show you how to fix it.
